### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR updates all dependencies in the project and big_tests. Also, rebar3 is upgraded.

Changes I needed to made due to upgraded versions of libraries:
* While generating config for big_tests when we want to enable some module we create a variable which is injected into template with `bbmustache`. In current version empty binary is considered falsey in conditions, so now when we want to enable module with no additional configuration we could inject a newline character.
* Deleting `users.csv.tmp` file after running `graphql_account_SUITE`
* Arguments for `init` function in `redis_client` module changed from list to proplist. Now we are creating them according to this.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/160261) / small_tests / 4b860f53b5d7b868896dfd12619b81fb76a3f9cf
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/160261/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/160261/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-01-05_12.30.06/lib.mongooseim.logs/run.2023-01-05_12.30.06/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/160336) / small_tests / 5bb1a4c6093a31e467c1d4c258218e7323c6f4bf
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/160336/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/160336/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-01-05_13.05.30/lib.mongooseim.logs/run.2023-01-05_13.05.30/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/160555) / small_tests / 5fa04c216995c15aa080e8488287e497cb9bd07c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/160555/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/160555/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-01-09_10.14.19/lib.mongooseim.logs/run.2023-01-09_10.14.20/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/160633) / small_tests / 7fe73339a7308e88377cf52e61ad60fa70d6a6da
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/160633/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/160633/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-01-09_11.10.09/lib.mongooseim.logs/run.2023-01-09_11.10.09/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/160663) / small_tests / 2f4d26d86fc15e34348214a87d99f0224222e11e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/160663/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/160663/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-01-09_11.55.21/lib.mongooseim.logs/run.2023-01-09_11.55.21/suite.log.html)

**Review by NelsonVides — CHANGES_REQUESTED** [review]
`rebar.lock` files are not updated, changes to the versions in the `rebar.config` files are ignored, what rebar really applies is what's written on the lockfile. You need to run `rebar3 upgrade --all` and rebar will actually install the new versions and carefully update the lockfile automatically 🙂

**Comment by mongoose-im** [comment]
[dynamic_domains_pgsql_mnesia_24](https://circleci.com/gh/esl/MongooseIM/161656) / pgsql_mnesia / 910e60e864fe475997127cc1367365b1fc924dff
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/161656/pgsql_mnesia.24.3.4)

**Comment by mongoose-im** [comment]
[dynamic_domains_mysql_redis_25](https://circleci.com/gh/esl/MongooseIM/164645) / mysql_redis / a18f10538404c1627caf03f4b270adb8aa5fe275
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/164645/mysql_redis.25.2)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/164778) / small_tests / bfd80388a1d20a59cc80dffaba248e1a0f3d27d6
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/164778/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/164778/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-01_12.16.40/lib.mongooseim.logs/run.2023-02-01_12.16.40/suite.log.html)

**Comment by mongoose-im** [comment]
[dynamic_domains_mysql_redis_25](https://circleci.com/gh/esl/MongooseIM/164880) / mysql_redis / 4bd9b7e10b5afa87c3a133828683299c58f7fdcc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/164880/mysql_redis.25.2)

**Comment by mongoose-im** [comment]
[dynamic_domains_mysql_redis_25](https://circleci.com/gh/esl/MongooseIM/164915) / mysql_redis / 815e4c4ab882627b84c7e891cc4eae05aac5539f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/164915/mysql_redis.25.2)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165187) / small_tests / ec71e65dd8ac848bead5e3c0012d421a0265f497
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/165187/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/165187/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-07_11.19.37/lib.mongooseim.logs/run.2023-02-07_11.19.37/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165236) / small_tests / 9ddfbfae0782cecd2abb464f0344d1ae6508542b
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/165236/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/165236/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-07_11.59.21/lib.mongooseim.logs/run.2023-02-07_11.59.21/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165259) / small_tests / de1a6e4017b70853f08c06d3706314ccf0a4b6f1
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/165259/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/165259/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-07_12.17.46/lib.mongooseim.logs/run.2023-02-07_12.17.47/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165294) / small_tests / 9c3a62229bf9b0e7e0dc6b544f49cfe796918384
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/165294/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/165294/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-07_12.34.54/lib.mongooseim.logs/run.2023-02-07_12.34.54/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165339) / small_tests / e18c41feb760c73a21d6ad9204e70ae728e9b488
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/165339/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/165339/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-07_13.02.46/lib.mongooseim.logs/run.2023-02-07_13.02.46/suite.log.html)

**Review by chrzaszcz — COMMENTED** [review]
Looks good. My only concern is https://github.com/esl/MongooseIM/blob/master/tools/test_runner/apply_templates.erl#L37

Does this line suffer from the same empty binary issue? It won't be caught by CI, because it is only used when you run tests locally with `test-runner.sh`

**Comment by pawlooss1** [comment]
> Looks good. My only concern is https://github.com/esl/MongooseIM/blob/master/tools/test_runner/apply_templates.erl#L37
> 
> Does this line suffer from the same empty binary issue? It won't be caught by CI, because it is only used when you run tests locally with `test-runner.sh`

I double-checked it and surprisingly when I add a similar line to this code tests fail because the nodes can't start. In this case such code is probably breaking the config. Without it all big_tests pass when run locally.

File: big_tests/rebar.config [hunk 1]
```diff
@@ -10,16 +10,16 @@
 {deps, [
         {lager, "3.9.2"},
         {base16, "2.0.1"},
-        {exml, "3.2.1", {pkg, hexml}},
-        {bbmustache, "1.10.0"},
+        {exml, "3.2.2", {pkg, hexml}},
```
Comment (NelsonVides) [inline big_tests/rebar.config:13]: There's a 3.3.3 one published 😬

**Review by NelsonVides — COMMENTED** [review]
Just a tiny little comment, same applies to both rebar.config files 😬

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165480) / small_tests / 687a772976b801898c534560d6f5137b8a8c706c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3918/165480/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3918/165480/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-09_13.02.51/lib.mongooseim.logs/run.2023-02-09_13.02.52/suite.log.html)

File: big_tests/rebar.config (hunk 1 above)
Comment (pawlooss1) [reply big_tests/rebar.config]: Done 😃 Thx for noticing. When I first looked at the newest versions it wasn't released yet.

**Review by NelsonVides — APPROVED** [review]
LGTM 👌🏽

**Pull request merged**